src/main/java/com/github/pedrovgs/androidwifiadb/view/AndroidDevices.java

```diff
@@ -0,0 +1,170 @@
+package com.github.pedrovgs.androidwifiadb.view;
+
+import com.github.pedrovgs.androidwifiadb.adb.ADB;
+import com.github.pedrovgs.androidwifiadb.adb.ADBParser;
+import com.github.pedrovgs.androidwifiadb.adb.AndroidWiFiADB;
+import com.github.pedrovgs.androidwifiadb.adb.CommandLine;
+import com.github.pedrovgs.androidwifiadb.model.Device;
+import com.intellij.notification.Notification;
+import com.intellij.notification.NotificationGroup;
+import com.intellij.notification.NotificationType;
+import com.intellij.notification.Notifications;
+import com.intellij.openapi.application.ApplicationManager;
+import com.intellij.openapi.project.Project;
+import com.intellij.openapi.wm.ToolWindow;
+import com.intellij.openapi.wm.ToolWindowFactory;
+import com.intellij.ui.content.Content;
+import com.intellij.ui.content.ContentFactory;
+
+import javax.swing.JPanel;
+
+/**
+ * Created by vgaidarji on 10/23/15.
+ */
+public class AndroidDevices implements ToolWindowFactory, View, DeviceAction {
+
+    private int INTERVAL_REFRESH_DEVICES = 1000;
+    private static final String ANDROID_WIFI_ADB_TITLE = "Android WiFi ADB";
+    private static final NotificationGroup NOTIFICATION_GROUP =
+            NotificationGroup.balloonGroup(ANDROID_WIFI_ADB_TITLE);
+
+    private final AndroidWiFiADB androidWifiADB;
+    private JPanel toolWindowContent;
+    private final CardLayoutDevices cardLayoutDevices;
+
+    public AndroidDevices() {
+        CommandLine commandLine = new CommandLine();
+        ADBParser adbParser = new ADBParser();
+        ADB adb = new ADB(commandLine, adbParser);
+        this.androidWifiADB = new AndroidWiFiADB(adb, this);
+        cardLayoutDevices = new CardLayoutDevices(toolWindowContent, this);
+    }
+
+    @Override
+    public void createToolWindowContent(Project project, ToolWindow toolWindow) {
+        this.androidWifiADB.updateProject(project);
+
+        ContentFactory contentFactory = ContentFactory.SERVICE.getInstance();
+        Content content = contentFactory.createContent(toolWindowContent, "", false);
+        toolWindow.getContentManager().addContent(content);
+
+        ApplicationManager.getApplication().executeOnPooledThread(new Runnable() {
+            public void run() {
+                setupUI();
+                monitorDevices();
+            }
+        });
+    }
+
+    private void setupUI() {
+        ApplicationManager.getApplication().invokeLater(new Runnable() {
+            @Override
+            public void run() {
+                cardLayoutDevices.createAndShowGUI();
+            }
+        });
+    }
+
+    /**
+     * Monitors connected devices state and triggers UI updates.
+     */
+    private void monitorDevices() {
+        ApplicationManager.getApplication().executeOnPooledThread(new Runnable() {
+            public void run() {
+                while (true) {
```

Comment:
We should use another implementation not based on an infinite loop and a ``Thread.sleep(....)`` method. Review the usage of java timers: https://docs.oracle.com/javase/tutorial/uiswing/misc/timer.html. If I'm not wrong we should use this class instead of this implementation.